app/forms/publish/user_form.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Publish
+  class UserForm < Form
+    FIELDS = %i[
+      first_name
+      last_name
+      email
+      id
+      code
+      authenticity_token
+    ].freeze
+
+    attr_accessor(*FIELDS)
+
+    validates :first_name, presence: true
+    validates :last_name, presence: true
+    validates :email, presence: true, format: { with: /\A.*@.*\z/, message: "Enter an email address in the correct format, like name@example.com" }
+    validate :email_is_lowercase
+
+    def provider_code_or_code(params)
+      params[:code] || params[:provider_code]
+    end
```

Comment:
This looks odd, can the correct thing be passed in the first time round?

```
UserForm.new(current_user, user, params: user_params, provider_code: provider_code)
```

Reply:
Yes, my routes were incorrect before so I was having issues with this. 

I've removed the helper method and just used params[:provider_code] here: https://github.com/DFE-Digital/publish-teacher-training/pull/3159/commits/ad8233a36d6fae55817fec06cae53e35f5005ccc